tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -66,6 +69,20 @@ final class ScalaJSClassEmitter private (
     linkingUnit.classDefs.map(c => c.encodedName -> c).toMap
   }
 
+  private[javascript] lazy val hasSubclasses: Set[String] = {
+    if (linkingUnit == null) {
+      throw new IllegalArgumentException(
+          "A class emitter created without a LinkingUnit cannot emit JS classes")
```

Comment:
That won't fly (what the error says is just not true). Either:

- Err on the safe side here when `linkingUnit` is null (will require not making it a set)
- Drop that deprecated ctor (+handling code) all together. We'll do it anyways in this release.